scipy/signal/spectral.py

```diff
@@ -511,11 +511,481 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+    >>> plt.ylim([0, f[-1]])
+    >>> plt.title('STFT Magnitude')
+    >>> plt.ylabel('Frequency [Hz]')
+    >>> plt.xlabel('Time [sec]')
+    >>> plt.show()
+    """
+
+    freqs, time, Zxx = _spectral_helper(x, x, fs, window, nperseg, noverlap,
+                                        nfft, detrend, return_onesided,
+                                        scaling='spectrum', axis=axis,
+                                        mode='stft', padded=padded,
+                                        centered=centered)
+
+    return freqs, time, Zxx
 
-    return freqs, time, Pxy
+
+def istft(Zxx, fs=1.0, window='hann', nperseg=None, noverlap=None, nfft=None,
+          input_onesided=True, centered=True, time_axis=-1, freq_axis=-2):
+    """
+    Perform the inverse Short Time Fourier transform (iSTFT).
+
+
+    Parameters
+    ----------
+    Zxx : array_like
+        STFT of the signal to be reconstructed. If a purely real array is
+        passed, it will be cast to a complex data type.
+    fs : float, optional
+        Sampling frequency of the time series. Defaults to 1.0.
+    window : str or tuple or array_like, optional
+        Desired window to use. See `get_window` for a list of windows and
+        required parameters. If ``window`` is array_like it will be used
+        directly as the window and its length will be used for nperseg.
+        Defaults to a Hann window. Must match the window used to generate
+        the STFT for faithful inversion.
+    nperseg : int, optional
+        Number of data points corresponding to each STFT segment. This
+        parameter must be specified if the number of data points per segment is
+        odd, or if the STFT was padded via ``nfft > nperseg``. If ``None``, the
+        value depends on the shape of ``Zxx`` and ``input_onesided``. If
+        ``input_onesided=True``, ``nperseg=2*(Zxx.shape[freq_axis] - 1)``.
+        Otherwise, ``nperseg=Zxx.shape[freq_axis]``.
+    noverlap : int, optional
+        Number of points to overlap between segments. If ``None``, half of the
+        segment length. Defaults to ``None``. When specified, the COLA constraint
+        must be met (see Notes below), and should match the parameter used to
+        generate the STFT.
+    nfft : int, optional
+        Number of FFT points corresponding to each STFT segment. This parameter
+        must be specified if the STFT was padded via ``nfft > nperseg``. If
+        ``None``, the default values are the same as for ``nperseg``, detailed
+        above, with one exception: if ``input_onesided`` is ``True`` and
+        ``nperseg==2*Zxx.shape[freq_axis] - 1``, ``nfft`` also takes on that
+        value. This case allows the proper inversion of an odd-length unpadded
+        STFT using ``nfft=None``.
+    input_onesided : bool, optional
+        If ``True``, interpret the input array as one-sided FFTs, such as is
+        returned by `numpy.fft.rfft`. If ``False``, interpret the input as a a
+        two-sided FFT.
```

Comment:
If we used `scipy.fftpack.rfft`, yes. `numpy.fft.rfft` uses complex numbers, however. Using the unpacked float arrays of fftpack were a huge headache to code robustly, and make plotting amplitudes and things like that much harder too.